lib/connections/connections.go

```diff
@@ -189,20 +219,20 @@ next:
 				limit := s.shouldLimit(c.Conn.RemoteAddr())
 
 				wr := io.Writer(c.Conn)
-				if limit && writeRateLimit != nil {
-					wr = &limitedWriter{c.Conn, writeRateLimit}
+				if limit && s.writeRateLimit != nil {
+					wr = &iolimiter.LimitedWriter{c.Conn, s.writeRateLimit}
 				}
 
 				rd := io.Reader(c.Conn)
-				if limit && readRateLimit != nil {
-					rd = &limitedReader{c.Conn, readRateLimit}
+				if limit && s.readRateLimit != nil {
+					rd = &iolimiter.LimitedReader{c.Conn, s.readRateLimit}
 				}
 
 				name := fmt.Sprintf("%s-%s (%s)", c.Conn.LocalAddr(), c.Conn.RemoteAddr(), c.Type)
 				protoConn := protocol.NewConnection(remoteID, rd, wr, s.model, name, deviceCfg.Compression)
```

Comment:
This protocol constructor expects a `model.Model` interface, which is why the `ConnectionHandlingModel` interface is larger than expected.

I haven't come up with any sane way around this, so far.